rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -727,45 +734,53 @@ TEST_F(TestNode, declare_parameter_with_cli_overrides) {
   // To match parameters YAML file content, use a well-known node name for this test only.
   auto node = std::make_shared<rclcpp::Node>("test_declare_parameter_node", no);
   {
-    rclcpp::ParameterValue value = node->declare_parameter("parameter_bool");
+    rclcpp::ParameterValue value = node->declare_parameter(
+      "parameter_bool", rclcpp::ParameterType::PARAMETER_BOOL);
     EXPECT_EQ(value.get_type(), rclcpp::PARAMETER_BOOL);
     EXPECT_EQ(value.get<bool>(), true);
   }
   {
-    rclcpp::ParameterValue value = node->declare_parameter("parameter_int");
+    rclcpp::ParameterValue value = node->declare_parameter(
+      "parameter_int", rclcpp::ParameterType::PARAMETER_INTEGER);
```

Comment:
I think we also need a test case for declaring an unset, typed parameter without an override. When I try this locally, I get a runtime exception. E.g.

    node->declare_parameter("my_param", rclcpp::PARAMETER_INTEGER);

without an override result in

```
terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterTypeException'
  what():  parameter 'my_param' has invalid type: Wrong parameter type, parameter {my_param} is of type {not set}, setting it to {not set} is not allowed.
```

I expected that this case shouldn't produce an exception.

Reply:
Thanks!

I knew I didn't test this case and I had a mental note about it, will fix.